nixos/modules/system/boot/networkd.nix

```diff
@@ -2995,13 +2998,18 @@ let
       systemd.sockets.systemd-networkd = {
         wantedBy = [ "initrd.target" ];
       };
+      systemd.services.systemd-resolved = {
+        wantedBy = ["initrd.target"];
+        serviceConfig.ExecStartPre = "-+/bin/ln -s /run/systemd/resolve/resolv.conf /etc/resolv.conf";
```

Comment:
I think this should be a tmpfiles rule, really.

Reply:
I have no idea how to do it properly. It would be better for you to upstream this